tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -1118,10 +1122,10 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
 
   private def lookupInfo(className: String)(
       implicit ctx: ErrorContext): Infos.ClassInfo = {
-    unit.infos.getOrElse(className, {
+    inputProvider.loadInfo(className).getOrElse {
       reportError(s"Cannot find info for class $className")
       Infos.ClassInfo(className)
-    })
+    }
```

Comment:
This part is still necessary because the IR checker needs the Infos of some classes that are not "needed at all", and are therefore not part of the `LinkingUnit`. An obvious example is abstract JS types that only appear in the signatures of methods. The IR checker wants to validate that `sjs_js_Function1` has type `any` when it appears in a signature, but the reachability analysis otherwise does not care about `js.Funtion1` (unless its `classOf` is needed for some reason).

Arguably, a future version of the Analyzer would at least *link* such classes, so that a `LinkingUnit` is internally consistent, and the IR checker does not need this "hack". Since the current PR does not make the situation better nor worse in this regard, this should be done later, if at all.